lib/shared-method.js

```diff
@@ -551,21 +558,34 @@ SharedMethod.toResult = function(returns, raw, ctx) {
 
   returns.forEach(function(item, index) {
     var name = item.name || item.arg;
-    if (convertToBasicRemotingType(item.type) === 'file') {
+    var targetType = convertToBasicRemotingType(item.type);
+    if (targetType === 'file') {
       console.warn('%s: discarded non-root return argument %s of type "file"',
         this.stringName,
         name);
       return;
     }
 
-    var value = convert(raw[index]);
+    var value = convert(raw[index], targetType);
     result[name] = value;
   });
 
   return result;
 
-  function convert(val) {
-    switch (SharedMethod.getType(val)) {
+  function convert(val, targetType) {
+    var targetTypeIsArray = Array.isArray(targetType) &&
+      targetType.length === 1;
+
+    if (targetTypeIsArray && Array.isArray(val)) {
+      var arrayTargetType = targetType[0];
+      return val.map(function(arg, ix) {
+        return convert(arg, arrayTargetType);
```

Comment:
I think this is a breaking change, because we were not converting date/buffer types in an array before. I.e. array items were serialized using default `JSON.serialize` convention before, this convention is different from what strong-remoting implements. See https://github.com/strongloop/loopback/issues/1990 and https://github.com/strongloop/loopback/issues/1907 for more context.

I agree the new behaviour is better, because it's what most people would expect, but I am not sure if we can back-port such change to 2.x.

I am proposing to rework `convert` method to ignore all arrays except "array of integers", in order to preserve backwards compatibility, and then address the issue of encoding of array items later.

@gunjpan @ritch thoughts?